Naming.md

```diff
@@ -46,3 +53,54 @@ protocol FooEventHandling {
     // good
 }
 ```
+
+### Naming a delegate method
+
+The first parameter should always be the item that the delegate was called for.
+
+#### Example:
+
+```swift
+// really bad: there's no parameter for the item that the delegate was called for
+func didDeleteDraft(draft: Draft)
+
+// bad: the item that the delegate was called for is not the first parameter
+func didDeleteDraft(draft: Draft, draftManager: DraftManager)
+
+// good
+func draftManager(_ manager: DraftManager, didDeleteDraft draft: Draft)
```

Comment:
Probably add an example similar to `func scrollViewDidScroll(_ scrollView: UIScrollView)`, when there's no parameters to pass, we put the action in the function name.